activerecord/test/cases/scoping/default_scoping_test.rb

```diff
@@ -142,6 +162,13 @@ def test_default_scope_with_all_queries_runs_on_update_columns
     assert_match(/mentor_id/, update_sql)
   end
 
+  def test_nilable_default_scope_with_all_queries_runs_on_update_columns
+    dev = DeveloperWithDefaultNilableMentorScopeAllQueries.create!(name: "Nikita")
+    update_sql = capture_sql { dev.update_columns(name: "Not Nikita") }.first
+
+    assert_no_match(/AND/, update_sql)
```

Comment:
That's a good point, the invalid `AND` statement is indeed at the end. Going to make the regex more specific 👍 

But to be honest now I feel like it would be better to check for `nothing_raised` instead. To make sure that we generate a valid statement in general, instead of checking for a very specific scenario. 
To be specific, @rafaelfranca would you prefer
```ruby
assert_no_match(/AND$/, update_sql)
```
which kind of checks that the specific issue has been fixed
or
```ruby
assert_nothing_raised { dev.update_columns(name: "Not Nikita") }
```
which looks like a "default scope generally works" test
Or perhaps both? 
Thanks!

Reply:
We avoid `assert_nothing_raised`. That is the same as calling `dev.update_columns(name: "Not Nikita")`, so we still need to assert something. Using `assert_nothing_raised` also doesn't assert if the reason why `update_columns` raised was because the issue is back. It could raise of unrelated reason, like a typo in `update_columns`. I think we need to assert the SQL.

Reply:
Done, changed the regex to check for an obscure `AND` only at the end 👍 
Thanks for clarifying!
Also rebased on the latest `main`, CI seems to be green ✅